Makefile

```diff
@@ -38,37 +43,45 @@ docs: ## generate Sphinx HTML documentation, including API docs
-	pip-compile --upgrade -o requirements/base.txt requirements/base.in
-	pip-compile --upgrade -o requirements/test.txt requirements/test.in
-	pip-compile --upgrade -o requirements/doc.txt requirements/doc.in
-	pip-compile --upgrade -o requirements/quality.txt requirements/quality.in
-	pip-compile --upgrade -o requirements/travis.txt requirements/travis.in
-	pip-compile --upgrade -o requirements/dev.txt requirements/dev.in
-	# Let tox control the Django version for tests
-	sed '/^[dD]jango==/d' requirements/test.txt > requirements/test.tmp
-	mv requirements/test.tmp requirements/test.txt
+# upgrade: export CUSTOM_COMPILE_COMMAND=make upgrade
+# upgrade: ## update the requirements/*.txt files with the latest packages satisfying requirements/*.in
+# 	pip install -qr requirements/pip-tools.txt
+# 	# Make sure to compile files after any other files they include!
+# 	pip-compile --upgrade -o requirements/pip-tools.txt requirements/pip-tools.in
+# 	pip-compile --upgrade -o requirements/base.txt requirements/base.in
+# 	pip-compile --upgrade -o requirements/test.txt requirements/test.in
+# 	pip-compile --upgrade -o requirements/doc.txt requirements/doc.in
+# 	pip-compile --upgrade -o requirements/quality.txt requirements/quality.in
+# 	pip-compile --upgrade -o requirements/travis.txt requirements/travis.in
+# 	pip-compile --upgrade -o requirements/dev.txt requirements/dev.in
+# 	# Let tox control the Django version for tests
+# 	sed '/^[dD]jango==/d' requirements/test.txt > requirements/test.tmp
+# 	mv requirements/test.tmp requirements/test.txt
 
 quality: ## check coding style with pycodestyle and pylint
-	tox -e quality
-
-requirements: ## install development environment requirements
-	pip install -qr requirements/pip-tools.txt
-	pip-sync requirements/dev.txt requirements/private.*
-
-test: clean ## run tests in the current virtualenv
-	pytest
+	pylint openedx_blockstore_relay $(PROJECT_ROOT)setup.py
+	pylint --py3k openedx_blockstore_relay $(PROJECT_ROOT)setup.py
+	pycodestyle $(PROJECT_ROOT)openedx_blockstore_relay $(PROJECT_ROOT)setup.py
+	isort --check-only --diff --recursive openedx_blockstore_relay $(PROJECT_ROOT)setup.py
+
+# requirements: ## install development environment requirements
+# 	pip install -qr requirements/pip-tools.txt
+# 	pip-sync requirements/dev.txt requirements/private.*
+
+# test target:
+# Meant to be run in Studio's vitrualenv
+# We don't use tox because we need to test with the real installation of
+# Open edX and whatever dependencies it uses.
+# We use no-cov and nomigrations because they _really_ speed up the test run.
+test: clean
+	python -m pytest --ds=cms.envs.test --no-cov --nomigrations $(PROJECT_ROOT)openedx_blockstore_relay/tests/
 
 diff_cover: test ## find diff lines that need test coverage
 	diff-cover coverage.xml
 
-test-all: ## run tests on every supported Python/Django combination
-	tox -e quality
-	tox
+# test-all: ## run tests on every supported Python/Django combination
```

Comment:
Do we want to remove the commented-out files?

Reply:
I have to keep `tox.ini` because it contains the `pycodestyle` settings etc. I can remove these few lines from the makefile though.